### DISCUSSION TITLE
How to ? any need ?

### DISCUSSION THREAD

**Discussion by luc-github** [luc-github/ESP3D#541] [Questions]
I would like to add some `How To?` Any suggestions ?

**Comment by DivingDuck** [comment]
Some ideas:
Maybe something like how to connect a specific board to an specific controller board and how to get it running. Some information's for hardware are already there but I think most of potential users have problems to get it running from setup perspective.

Other howto's are maybe something like how can someone integrate additional sensors, custom extensions, buttons etc.

**Reply by luc-github** [reply]
when you mean setup, do you mean like https://github.com/luc-github/ESP3D/discussions/542 ?

>Other howto's are maybe something like how can someone integrate additional sensors, custom extensions, buttons etc.

This is at API level - it would need  more than a little how to  I think

**Reply by Overnuts** [reply]
BigTreeTech embeds your work in a small wifi plugin for their SKR-PRO and TFT35 devices.
Right now, I'm trying to set up a running SKR-pro + tft35 + wifi  on the MPCNC...
I succed to upgrade esp3d.bin and esp3d-webui, set it up to reach my LAN and let firefox talk to it.
But it speaks to nobody....
A HOWTO "make mpcnc move from firefox" would be nice  😃

**Reply by luc-github** [reply]
@Overnuts I do not have all hardware so I may not be able to cover all `how to` like this

If you have issue to setup,  I think the best is first to open question to describe your setup issue / question, then we can work on solving it, once solution is found - a clear `how to` can then be done

**Reply by luc-github** [reply]
did you checked : https://www.youtube.com/watch?v=Z5Cnm-1ybFY
and https://www.youtube.com/watch?v=NGgzw-XayEo
both explain Marlin setup for SKR Pro - I do not know what you use on your MPCNC

**Reply by DivingDuck** [reply]
@luc-github Yes, that came into my mind.

**Reply by Overnuts** [reply]
> @Overnuts I do not have all hardware so I may not be able to cover all `how to` like this
> 
> If you have issue to setup, I think the best is first to open question to describe your setup issue / question, then we can work on solving it, once solution is found - a clear `how to` can then be done

Hi, I understand, you can't have all devices !
I found a few explanations here (fr) https://www.civade.com/post/2020/09/22/Ajouter-le-support-du-Wifi-sur-une-imprimante-3D-fonctionnant-avec-Marlin-2
At the end of the page "fixup" part
I should try this, and let you know, or there is a way to do in the setup ?
Do you want me to add an issue to ask for a parameter like "on boot, wait  xx seconds"  ?

**Reply by luc-github** [reply]
@Overnuts but as you see in fixup, this parameter already exists in 2.1 `myesp3d.begin(15000); // jpc wait 15 seconds`
in 3.0 there is a command and setting to change it without recompiling
